src/hotspot/share/gc/parallel/parallelScavengeHeap.cpp

```diff
@@ -401,15 +401,15 @@ ParallelScavengeHeap::death_march_check(HeapWord* const addr, size_t size) {
 HeapWord* ParallelScavengeHeap::mem_allocate_old_gen(size_t size) {
   if (!should_alloc_in_eden(size) || GCLocker::is_active_and_needs_gc()) {
     // Size is too big for eden, or gc is locked out.
-    return old_gen()->allocate(size);
+    return old_gen()->allocate_and_record(size);
```

Comment:
I would have kind of preferred if `allocate_and_record` were a helper method here in `ParallelScavengeHeap` since the recording seems to be entirely a thing of the PSH and not of the old gen, and the implementation of that method just calls back in here, but I am good with this too.